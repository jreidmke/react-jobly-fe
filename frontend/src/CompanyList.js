import Search from './Search';
import CompanyCard from './CompanyCard';
import { useEffect, useState, useContext } from 'react';
import {useHistory} from 'react-router-dom';
import JoblyApi from './api';
import UserContext from './UserContext'

const CompanyList = () => {
    const history = useHistory();
    const user = useContext(UserContext);

    if(!user.currUser) {
        history.push('/');
    }

    const [companies, setCompanies] = useState();

    useEffect(() => {
        async function getAllCompanies() {
            let comRes = await JoblyApi.getAllCompanies();
            setCompanies(comRes);
        }
        getAllCompanies();
    }, []);

    async function search(formData) {
        const searchRes = await JoblyApi.getCompanies(formData);
        setCompanies(searchRes);
    } 
    return(
        <div>
            <Search handleSearch={search}/>
            {companies ? companies.map(c => (
                <CompanyCard name={c.name} description={c.description} logo={c.logoUrl} handle={c.handle}/> 
            )) : "Loading"}

        </div>
    )
}

export default CompanyList;