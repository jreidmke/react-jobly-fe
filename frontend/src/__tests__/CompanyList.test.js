import { render, fireEvent, waitFor } from '@testing-library/react';
import CompanyList from '../CompanyList';
import { MemoryRouter } from "react-router";
import UserContext from '../UserContext';

const currUser = {username: "jreid", firstName: "James", lastName: "Reid", email: "jreidmke@gmail.com", isAdmin: false, applcations: []};

it("renders without crashing", () => {
    render(
        <UserContext.Provider value={currUser}>
            <MemoryRouter>
                <CompanyList />
            </MemoryRouter>
        </UserContext.Provider>
    )
});

it("matches snapshot", () => {
    const {asFragment} = render(
        <UserContext.Provider value={currUser}>
            <MemoryRouter>
                <CompanyList />
            </MemoryRouter>
        </UserContext.Provider>        
    );
    expect(asFragment()).toMatchSnapshot();
})

it("returns correct companies when searched", async() => {
    const { getByPlaceholderText, getByText, queryByText } = render(
        <UserContext.Provider value={currUser}>
            <MemoryRouter>
                <CompanyList/>
            </MemoryRouter>
        </UserContext.Provider>
    );

    //Loading should be displayed on page on first render
    const loading = queryByText("Loading");
    expect(loading).toBeInTheDocument();

    //Loading should be removed by time of company list load/search
    const button = getByText("Search");
    fireEvent.click(button);
    // waitFor(() => expect(loading).not.toBeInTheDocument());

    // //Search should return companies which match parameters
    // const hallDavis = waitFor(() => queryByText("Hall-Davis"));
    // const input = getByPlaceholderText("Search Here");
    // fireEvent.change(input, { target: { value: 'ba' }});
    // fireEvent.click(button);
    // const baker = waitFor(() => queryByText("Baker-Santos"));
    // waitFor(() => expect(baker).toBeInTheDocument());
    // waitFor(() => expect(hallDavis).not.toBeInTheDocument());
})

