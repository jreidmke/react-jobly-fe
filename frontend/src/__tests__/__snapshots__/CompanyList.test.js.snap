// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        id="search-term"
        name="searchTerm"
        placeholder="Search Here"
        type="text"
      />
      <button>
        Search
      </button>
    </form>
    Loading
  </div>
</DocumentFragment>
`;
